File: solutions/typescript/collatz-conjecture/1/collatz-conjecture.ts
export function steps(count: number): number {
  if (count <= 0) throw new Error("Only positive numbers are allowed");
  
  let result = 0;
  
  while (count > 1){
    if (count % 2 == 0){
      count /= 2;
    } else {
      count = 3 * count + 1;
    }
    result++;
  }

  return result;
}

File: solutions/python/collatz-conjecture/1/collatz_conjecture.py
def steps(number: int ) -> int:
    if number < 1:
        raise ValueError("Only positive integers are allowed")
    result: int = 0
    while number > 1: 
        result += 1
        if number % 2 == 0:
            number /= 2
        else:
            number = (3 * number) + 1
    return result